pkg/asset/installconfig/installconfig.go

```diff
@@ -0,0 +1,84 @@
+package installconfig
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/pborman/uuid"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// InstallConfig generates the install-config.yml file.
+type installConfig struct {
+	assetStock  Stock
+	directory   string
+	inputReader *bufio.Reader
+}
+
+var _ asset.Asset = (*installConfig)(nil)
+
+// Dependencies returns all of the dependencies directly needed by an
+// installConfig asset.
+func (a *installConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		a.assetStock.EmailAddress(),
+		a.assetStock.Password(),
+		a.assetStock.BaseDomain(),
+		a.assetStock.ClusterName(),
+		a.assetStock.License(),
+		a.assetStock.PullSecret(),
+		a.assetStock.Platform(),
+	}
+}
+
+// Generate generates the install-config.yml file.
+func (a *installConfig) Generate(dependencies map[asset.Asset]*asset.State) (*asset.State, error) {
+	clusterName := string(dependencies[a.assetStock.ClusterName()].Contents[0].Data)
+	platform := string(dependencies[a.assetStock.Platform()].Contents[0].Data)
+
+	installConfig := types.InstallConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterName,
+		},
+		ClusterID: uuid.NewUUID(),
+	}
+
+	switch platform {
+	case AWSPlatformType:
+		installConfig.AWS = &types.AWSPlatform{}
+	case LibvirtPlatformType:
+		installConfig.Libvirt = &types.LibvirtPlatform{}
+	default:
+		return nil, fmt.Errorf("unknown platform type %q", platform)
+	}
+
+	data, err := yaml.Marshal(installConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := os.MkdirAll(a.directory, 0755); err != nil {
+		return nil, err
+	}
+	assetPath := filepath.Join(a.directory, "install-config.yml")
```

Comment:
I think many packages will need to perform this. so either move it to a common helpfer func (ideally it can just take the `asset.State` and iterate over the `contents`) or make the `Store` to do it? But I am aware that some targets won't need to write stuff into disk (e.g. the targetable assets like `Cluster`)

Reply:
> But I am aware that some targets won't need to write stuff into disk (e.g. the targetable assets like Cluster)

I think we should just write all the assets to disk.  It should help with reloading from disk for regeneration (or for generating the next asset group).  And even if an asset will never be needed again, disk space isn't that expensive ;)

Reply:
@wking I know, by `targetable assets` I mean things like `Cluster`, `Manifests` which are not the actual assets, they just represent a group of assets, so they don't have any content to write.

Reply:
> ... by `targetable assets` I mean things like `Cluster`, `Manifests` which are not the actual assets, they just represent a group of assets, so they don't have any content to write.

I'd rather not create pseudo-assets for grouping.

Reply:
that's already part of the dependency graph, it's meant to be the entrypoint for different stages of the installer, so that users can insert their customization.
The installer's subcommands (`openshift-install installconfig`, `openshift-install ignition`) also correspond to these targets.